app/helpers/dashboard_helper.rb

```diff
@@ -0,0 +1,29 @@
+module DashboardHelper
+  def stack
+    Faraday::RackBuilder.new do |builder|
+      builder.use Faraday::HttpCache, serializer: Marshal, shared_cache: false
+      builder.use Octokit::Response::RaiseError
+      builder.adapter Faraday.default_adapter
+    end
+  end
+
+  def client
+    Octokit.middleware = stack
+    client = Octokit::Client.new(user: current_user.github_login, access_token: current_user.token)
+    client.auto_paginate = true
+    client
+  end
+
+  def current_client
+    @current_client ||= client
+  end
+
+  def gist_action(action)
+    respond_to do |format|
+        format.html { redirect_to Octokit.gist(params[:id]).url }
+        format.js {
+          render json: { error: "Couldn't #{action} the gist" } unless current_client.send("#{action}_gist", params[:id])
```

Comment:
Maybe we need to add some efficient error message.

Reply:
My thoughts exactly! However, I don't have any idea how to do it.

If you check [#star_gist](http://octokit.github.io/octokit.rb/Octokit/Client/Gists.html#star_gist-instance_method) documentation, this method only returns a boolean value.  Instead of just returning `true` or `false` I've added that `render json: { error: "Couldn't #{action} the gist" }` part.

I am open to any suggestion to make it better.

Reply:
You're right. API returns bool in response. I will think about how to handle it